callback_plugins/default_plus_summary.py

```diff
@@ -0,0 +1,94 @@
+# vim: expandtab:tabstop=4:shiftwidth=4
+'''
+Ansible callback plugin.
+'''
+
+from ansible.plugins.callback.default import CallbackModule as CallbackModule_default
+from ansible import constants as C
+from ansible.utils.color import stringc
+
+
+class CallbackModule(CallbackModule_default):
+    '''
+    This is like the default callback plugin, but also stores results and
+    summarizes failures.
+    '''
+
+    CALLBACK_VERSION = 2.0
+    CALLBACK_TYPE = 'stdout'
```

Comment:
Trying to get it to work as an 'aggregate' plugin, got this error.
@detiber have you seen this before?

While I understand the Python part of the error, I fail to understand what Ansible is doing...


```console
$ ANSIBLE_DEBUG=1 ansible-playbook -i hosts.ini playbooks/adhoc/preflight/check.yml -vvv
  9855 1483635797.56724: starting run
Using /home/vagrant/src/github.com/openshift/openshift-ansible/ansible.cfg as config file
  9855 1483635797.69092: Loading CacheModule 'memory' from /usr/lib/python2.7/site-packages/ansible/plugins/cache/memory.py
  9855 1483635797.84638: Loading CallbackModule 'default_plus_summary' from /home/vagrant/src/github.com/openshift/openshift-ansible/callback_plugins/default_plus_summary.py
  9855 1483635797.84912: Loading CallbackModule 'aa_version_requirement' from /home/vagrant/src/github.com/openshift/openshift-ansible/callback_plugins/aa_version_requirement.py (found_in_cache=False, class_only=True)
  9855 1483635797.84947: Loading CallbackModule 'actionable' from /usr/lib/python2.7/site-packages/ansible/plugins/callback/actionable.py (found_in_cache=False, class_only=True)
  9855 1483635797.84981: Loading CallbackModule 'context_demo' from /usr/lib/python2.7/site-packages/ansible/plugins/callback/context_demo.py (found_in_cache=False, class_only=True)
  9855 1483635797.85005: Loading CallbackModule 'debug' from /usr/lib/python2.7/site-packages/ansible/plugins/callback/debug.py (found_in_cache=False, class_only=True)
  9855 1483635797.85207: Loading CallbackModule 'default' from /home/vagrant/src/github.com/openshift/openshift-ansible/callback_plugins/default.py (found_in_cache=False, class_only=True)
  9855 1483635797.85252: Loading CallbackModule 'default' from /usr/lib/python2.7/site-packages/ansible/plugins/callback/default.py (found_in_cache=False, class_only=True)
  9855 1483635797.85268: Loading CallbackModule 'default_plus_summary' from /home/vagrant/src/github.com/openshift/openshift-ansible/callback_plugins/default_plus_summary.py (found_in_cache=False, class_only=True)
  9855 1483635797.85284: RUNNING CLEANUP
ERROR! Unexpected Exception: super(type, obj): obj must be an instance or subtype of type
the full traceback was:

Traceback (most recent call last):
  File "/usr/bin/ansible-playbook", line 103, in <module>
    exit_code = cli.run()
  File "/usr/lib/python2.7/site-packages/ansible/cli/playbook.py", line 159, in run
    results = pbex.run()
  File "/usr/lib/python2.7/site-packages/ansible/executor/playbook_executor.py", line 89, in run
    self._tqm.load_callbacks()
  File "/usr/lib/python2.7/site-packages/ansible/executor/task_queue_manager.py", line 195, in load_callbacks
    self._callback_plugins.append(callback_plugin())
  File "/home/vagrant/src/github.com/openshift/openshift-ansible/callback_plugins/default_plus_summary.py", line 22, in __init__
    super(CallbackModule, self).__init__()
  File "/usr/lib/python2.7/site-packages/ansible/plugins/callback/default.py", line 41, in __init__
    super(CallbackModule, self).__init__()
TypeError: super(type, obj): obj must be an instance or subtype of type
```


The diff to the code here is just:

```diff
diff --git a/callback_plugins/default_plus_summary.py b/callback_plugins/default_plus_summary.py
index 610a3e9..77452dc 100644
--- a/callback_plugins/default_plus_summary.py
+++ b/callback_plugins/default_plus_summary.py
@@ -15,7 +15,7 @@ class CallbackModule(CallbackModule_default):
     '''
 
     CALLBACK_VERSION = 2.0
-    CALLBACK_TYPE = 'stdout'
+    CALLBACK_TYPE = 'aggregate'
     CALLBACK_NAME = 'default_plus_summary'
 
     def __init__(self):
```